Tools/Algorithms/parser.py

```diff
@@ -0,0 +1,31 @@
+def parse_input(input_file):
+    """
+    Parse WarpX input file.
+
+    Parameters
+    ----------
+    input_file : string
+        Path to input file.
+
+    Returns
+    -------
+    input_dict : dictionary
+        Dictionary storing WarpX input parameters
+        (parameter's name stored as key, parameter's value stored as value).
+    """
+    input_dict = dict()
+    for line in open(input_file):
+        sline = line.split('=')
+        # skip lines that are commented out or blank
+        skip_line = sline[0].startswith('#') or sline[0].startswith('\n')
+        if not skip_line:
+            key = sline[0].strip()
+            val = sline[1].split()
+            # The value corresponding to a given key of input_dict is a list
+            # of strings, from which we remove any leftover comments
+            for w in val:
```

Comment:
Modifying the iterator in a loop is not good practice. In this case, if `val` has two elements that start with '#', the loop crashes with a ValueError. This can be avoided by adding a `break`. Here's an alternative, somewhat simpler version
```
for i in range(len(val)):
    if val[i].startswith('#'):
        val = val[:i]
        break
```